### PR TITLE
Update Ginkgo to v2

### PR DESCRIPTION
**Pull request by hawkowl**
### What this PR does / why we need it:

Gives us some extra tools for combating flaky tests. See https://onsi.github.io/ginkgo/MIGRATING_TO_V2#flakeattempts-decorator

### Test plan for issue:

Should pass E2E, since it's just an E2E update.

### Is there any documentation that needs to be updated for this PR?

N/A

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -55,7 +55,7 @@ require (
 	github.com/stretchr/testify v1.7.1
 	github.com/ugorji/go/codec v1.2.7
```
Comment (m1kola) [inline go.mod:56]: `go.mod` still uses `github.com/onsi/ginkgo` of version ` v1.16.5`. Or do I miss something?

Comment (hawkowl) [reply go.mod:56]: UGH I FORGOT ABOUT GO'S SILLY /V2 THING

<fixes>

File: .gitignore [hunk 2]
```diff
@@ -11,6 +11,7 @@ gomock_reflect_*
 /aro
 /dev-config.yaml
 /e2e.test
+/test/e2e/e2e.test
```
Comment (m1kola) [inline .gitignore:14]: Trying to rebase this branch right now. Not sure that I understand why we need this one.

I'll remove it for now to see what is the difference. Probably it is something CI/docker image related and I do not see issues in my locals dev workflow.

File: .pipelines/templates/template-prod-e2e-steps.yml [hunk 3]
```diff
@@ -61,6 +61,6 @@ steps:
       --env CLUSTER="$CLUSTER" \
       --entrypoint e2e.test \
       "$IMAGE:$VERSION" \
-      -test.timeout 180m -test.v -ginkgo.v
+      -ginkgo.timeout 180m -test.v -ginkgo.v
```
Comment (m1kola) [inline .pipelines/templates/template-prod-e2e-steps.yml:64]: Do we still use this yaml given that we migrated to EV2?

File: deps.go [hunk 4]
```diff
@@ -14,6 +15,7 @@ import (
 	_ "github.com/golangci/golangci-lint/cmd/golangci-lint"
 	_ "github.com/jewzaam/go-cosmosdb/cmd/gencosmosdb"
 	_ "github.com/jstemmer/go-junit-report"
+	_ "github.com/onsi/ginkgo/v2/ginkgo"
```
Comment (m1kola) [inline deps.go:18]: Wondering why did the authors of ginkgo decide to have this `/v2` prefix instead of releasing new major version?

**Review by m1kola — COMMENTED** [review]
A tip just in case we need to rebase it again: if you see something like `flag redefined: ginkgo.seed` - this is like mean that on rebase you got new in master tests which import ginkgo v1 (`github.com/onsi/ginkgo`) while tests in this PR already have Ginkgo v2.

Solution is to make sure that all imports of `github.com/onsi/ginkgo` where replaced with `github.com/onsi/ginkgo/v2`

**Comment by m1kola** [comment]
> Gives us some extra tools for combating flaky tests. See https://onsi.github.io/ginkgo/MIGRATING_TO_V2#flakeattempts-decorator

@hawkowl v1 has `flakeAttempts` flag which allows to specify this once for all the tests. I added this flag in this PR: https://github.com/Azure/ARO-RP/pull/2416

I still think that we need to update. Your changes look good (the only this which was not clear to me is [this](https://github.com/Azure/ARO-RP/pull/2180#discussion_r981148147)). I rebased your branch on top of master and will merge on green E2E.

**Review by m1kola — APPROVED** [review]
Everything looks good and CI seems to be all green ✅

**Pull request merged**